File: app/src/main/java/io/nanodegree/andrea/popularmovies/model/VideoContainer.java
package io.nanodegree.andrea.popularmovies.model;

import com.squareup.moshi.Json;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

/**
 * Created by Andrea Loddo (andrea@evenly.io) on 20.12.2018
 * for Evenly GmbH
 * <p>
 * Copyright (c) 2018 Evenly GmbH,
 * all rights reserved
 */
public class VideoContainer {

    private static final String TRAILER_KEYWORD = "trailer";
    private static final String YOUTUBE_KEYWORD = "youtube";

    @Json(name = "results")
    private List<Video> videoList;

    public List<Video> getYoutubeTrailers() {

        List<Video> tempVideoList = new ArrayList<>(videoList);

        if (tempVideoList.size() > 0) {

            for (Iterator<Video> videoIterator = videoList.listIterator(); videoIterator.hasNext(); ) {
                Video nextVideo = videoIterator.next();

                if (!nextVideo.type.contains(TRAILER_KEYWORD) || !nextVideo.site.contains(YOUTUBE_KEYWORD)) {
                    videoIterator.remove();
                }
            }

        }

        return tempVideoList;
    }
}

File: app/src/main/java/io/nanodegree/andrea/popularmovies/data/model/VideoContainer.kt
package io.nanodegree.andrea.popularmovies.data.model

import com.google.gson.annotations.SerializedName
import java.util.*

class VideoContainer(@SerializedName("results")
                     private val videoList: MutableList<Video>? = null) {

    fun getYoutubeTrailers(): List<Video>? {

        val tempVideoList = ArrayList(videoList!!)

        if (tempVideoList.size > 0) {

            val videoIterator = videoList.listIterator()
            while (videoIterator.hasNext()) {
                val (_, _, _, site, type) = videoIterator.next()

                if (!(type == TRAILER_KEYWORD && site == YOUTUBE_KEYWORD)) {
                    videoIterator.remove()
                }
            }
        }

        return tempVideoList
    }

    companion object {
        private const val TRAILER_KEYWORD = "Trailer"
        private const val YOUTUBE_KEYWORD = "YouTube"
    }
}
